taichi/transforms/lower_ast.cpp

```diff
@@ -204,9 +204,8 @@ class LowerAST : public IRVisitor {
       auto end = stmt->end;
       flatten_rvalue(begin, &fctx);
       flatten_rvalue(end, &fctx);
-      bool is_good_range_for =
-          current_block_depth_ == 1 || detected_fors_with_break_.find(stmt) ==
-                                           detected_fors_with_break_.end();
+      bool is_good_range_for = detected_fors_with_break_.find(stmt) ==
```

Comment:
Why this change? I forget what is the behavior of top-level break now...

Reply:
We are now supporting serial fors at the outermost scope, and they can contain break statements, so we should convert them to while statements.